pghoard/basebackup.py

```diff
@@ -419,7 +442,16 @@ def write_files_to_tar(self, *, files, tar):
                 continue
 
             try:
-                tar.add(local_path, arcname=archive_path, recursive=False)
+                if delta_stats is None:
+                    tar.add(local_path, arcname=archive_path, recursive=False)
+                else:
+                    if os.path.isdir(local_path):
+                        tar.add(local_path, arcname=archive_path, recursive=False)
+                    else:
+                        with HashFile(path=local_path) as fileobj:
+                            ti = tar.gettarinfo(name=local_path, arcname=archive_path)
+                            tar.addfile(ti, fileobj=fileobj)
+                            delta_stats[fileobj.hash.hexdigest()] = ti.size
```

Comment:
hmm, looks a little bit too nested for me, maybe this way it reads better:
```
if os.path.isdir(local_path):
    tar.add(....
    continue

if delta_stats is None:
    tar.add(...
else
    with HashFile(...
```

Reply:
It's not the same, we care about if path is a dir only in case of delta_stats is specified.
The first `if` just follows the old logics, the `else` thingy is for delta statistics.

Reply:
Logically it is, we add directories anyway (stats or no stats), and for other paths we have stats-dependent behaviour. Although checking `isdir` is more expensive that just checking for `None`, and we need to check only when stats are required, so keeping this code as it is might be better.

Reply:
ah, true, I am thinking too much about my case :)